node/service/src/lib.rs

```diff
@@ -614,16 +614,31 @@ where
 	let overrides = crate::rpc::overrides_handle(client.clone());
 	let fee_history_limit = rpc_config.fee_history_limit;
 
-	rpc::spawn_essential_tasks(rpc::SpawnTasksParams {
-		task_manager: &task_manager,
-		client: client.clone(),
-		substrate_backend: backend.clone(),
-		frontier_backend: frontier_backend.clone(),
-		filter_pool: filter_pool.clone(),
-		overrides: overrides.clone(),
-		fee_history_limit,
-		fee_history_cache: fee_history_cache.clone(),
-	});
+	// Sinks for pubsub notifications.
+	// Everytime a new subscription is created, a new mpsc channel is added to the sink pool.
+	// The MappingSyncWorker sends through the channel on block import and the subscription emits a
+	// notification to the subscriber on receiving a message through this channel.
+	// This way we avoid race conditions when using native substrate block import notification
+	// stream.
+	let pubsub_notification_sinks: fc_mapping_sync::EthereumBlockNotificationSinks<
+		fc_mapping_sync::EthereumBlockNotification<Block>,
+	> = Default::default();
+	let pubsub_notification_sinks = Arc::new(pubsub_notification_sinks);
```

Comment:
A question about the channel, if this is created per-subscription is it possible to OOM kill the client via flooding it with subscriptions?

Reply:
If the connection is kept open in the client end, the limit of channels is the number of websocket connections the server supports. Each time a new ethereum block is produced, channels associated to closed connections are removed from the pool.